File: backend/src/Organi.Server/Organi.Server.Domain/Entities/Coupon.cs
using Organi.Server.Domain.Common;
using Organi.Server.Domain.Enums;

namespace Organi.Server.Domain.Entities;

public sealed class Coupon : AuditableEntity
{
    public string Code { get; set; } = default!;
    public string? Description { get; set; }
    public DiscountType DiscountType { get; set; }
    public decimal DiscountValue { get; set; }
    public decimal? MinimumOrderAmount { get; set; }
    public int? MaxUsageCount { get; set; }
    public int CurrentUsageCount { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public bool IsActive { get; set; } = true;

    public ICollection<Order> Orders { get; set; } = [];
}

File: frontend/admin/src/types/api/coupon.ts
// Mirrors Organi.Server.Application.Features.Coupons.DTOs.CouponResponse
export interface CouponResponse {
  id: string
  code: string
  description: string | null
  discountType: 'Percentage' | 'FixedAmount'
  discountValue: number
  minimumOrderAmount: number | null
  maxUsageCount: number | null
  currentUsageCount: number
  startDate: string
  endDate: string
  isActive: boolean
  createdAt: string
  updatedAt: string | null
}

// Mirrors CreateCouponCommand
export interface CreateCouponRequest {
  code: string
  description: string | null
  discountType: string
  discountValue: number
  minimumOrderAmount: number | null
  maxUsageCount: number | null
  startDate: string
  endDate: string
}

// Mirrors UpdateCouponCommand (Id travels in the URL, not the body)
export interface UpdateCouponRequest extends CreateCouponRequest {
  isActive: boolean
}
